### ISSUE TITLE
Running the artisynth on macos catalina 10.15.6, fails

### ISSUE THREAD

**Issue by amirnn** [artisynth/artisynth_core#9]
I am attaching the report as a text, but the short message I receive when using zsh trying to run artisynth is below. I tried running with both current development version and with the lastest stable release provided on the website.
I am using java and javac version "1.8.0_261".

2020-07-17 00:19:26.773 java[5801:136441] Apple AWT Internal Exception: NSWindow drag regions should only be invalidated on the Main Thread!
2020-07-17 00:19:26.773 java[5801:136441] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow drag regions should only be invalidated on the Main Thread!'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff316b4b57 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6a4fb5bf objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff316dd34c -[NSException raise] + 9
	3   AppKit                              0x00007fff2e8d75ec -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 310
	4   AppKit                              0x00007fff2e8bf052 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1416
	5   AppKit                              0x00007fff2e8beac3 -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
	6   libnativewindow_macosx.jnilib       0x000000012cfd73fe Java_jogamp_nativewindow_macosx_OSXUtil_CreateNSWindow0 + 398
	7   ???                                 0x000000010a0eb6c7 0x0 + 4463703751
)
libc++abi.dylib: terminating with uncaught exception of type NSException
./bin/artisynth: line 188:  5801 Abort trap: 6           java $JAVA_OPTS artisynth.core.driver.Launcher "${MAIN_OPTS[@]}"

**Comment by JohnLloyd7** [comment]
Hi Amir, thanks for the report. The problem has to do with the Java OpenGL (JOGL) bindings, which we rely on to provide our 3D graphics. The error you describe had been appearing as a warning in some MacOS releases prior to Catalina, but now (at least on your hardware) seems to be throwing an error. The error is reported to have been fixed in the latest version of JOGL (2.4), but that is not yet officially released, which is why it is not included in ArtiSynth. I'll try obtaining a test version of 2.4 and see if I can make it available on an as-needed basis. The error may also be hardware dependent. If you can let me know exactly what Mac hardware you have, that might be helpful.

**Comment by amirnn** [comment]
Hi John, thank you for your fast reply. My hardware is:

Model: MacBookPro13,3, BootROM 428.0.0.0.0, 4 processors, Quad-Core Intel Core i7, 2.9 GHz, 16 GB, SMC 2.38f11
Graphics: kHW_IntelHDGraphics530Item, Intel HD Graphics 530, spdisplays_builtin
Graphics: kHW_AMDRadeonPro460Item, AMD Radeon Pro 460, spdisplays_pcie_device, 4 GB
Memory Module: BANK 0/DIMM0, 8 GB, LPDDR3, 2133 MHz, 0x802C, 0x4D5435324C31473332443450472D30393320
Memory Module: BANK 1/DIMM0, 8 GB, LPDDR3, 2133 MHz, 0x802C, 0x4D5435324C31473332443450472D30393320

**Comment by JohnLloyd7** [comment]
Hi Amir, I obtained the latest version of JOGL 2.4.0 (which is not yet officially released) and was able to compile ArtiSynth with it. When I run on my (older) MacBook Air, which has Catalina installed, the "NSWindow drag regions ..." warning goes away (I only got a warning on Catalina, not a error as you did). Anyway, try it on your machine. To change to JOGL 2.4.0,
edit <ARTISYNTH_HOME>/lib/LIBRARIES and change the lines 

gluegen-rt-2.3.2.jar
jogl-all-2.3.2.jar
MacOS gluegen-rt-2.3.2-natives-macosx-universal.jar
MacOS jogl-all-2.3.2-natives-macosx-universal.jar

to

gluegen-rt-2.4.0.jar
jogl-all-2.4.0.jar
MacOS gluegen-rt-2.4.0-natives-macosx-universal.jar
MacOS jogl-all-2.4.0-natives-macosx-universal.jar

Then remove the 2.3.2 jar files in <ARTISYNTH_HOME>/lib

> rm *2.3.2*.jar

and run <ARTISYNTH_HOME>/bin/updateArtisynthLibs to download to the 2.4.0 files. Then recompile ArtiSynth and run it. You might get an "invalid framebuffer operation" message - but that might be benign.

John

**Comment by JohnLloyd7** [comment]
Hello again,
As I hoped, the "invalid framebuffer operation" message appears to occur only when checking what GLVersions are supported and appears to be benign, so I have pushed a change to suppress this error. After I get more information about how widespread the JOGL/Catalina issue is, I will look at maybe supplying the pre-release JOGL 2.4.0 for MacOS. Meanwhile, you can change to JOGL 2.4.0 as described in my previous message. Note that if you are using the Eclipse IDE, you will also need to edit the artisynth_core build path and replace the libraries gluegen-rt-2.3.2.jar and jogl-all-2.3.2.jar with gluegen-rt-2.4.0.jar and jogl-all-2.4.0.jar. (The "natives" jar files do not appear in the Eclipse build path and so there's nothing to change there.)

**Comment by amirnn** [comment]
Dear John,

Thank you for your response and excuse me for my late and delayed recheck. I tired the fixes you had suggested, however, the GUI seems to not show some of the options and it is until a mouse hover happens that they appear. Also, no canvas is displayed. In the meantime, I am using the ARTISYNTH on a linux environment. 

Thank you and Best regards, Amir

**Comment by kkarmst** [comment]
Hi John,

For others on macOS Catalina 10.15.6 having the same issue as Amir, I followed the advice to change the specified libraries to JOGL 2.4.0 in <ARTISYNTH_HOME>/lib/LIBRARIES and I was able to compile then launch the GUI with no warnings or errors from the command line.

No issues with the GUI. I am using javac and java "1.8.0_261".

**Hardware:**
Model Name:	MacBook Air
Model Identifier:	MacBookAir8,2
Processor Name:	Dual-Core Intel Core i5
Processor Speed:	1.6 GHz
Number of Processors:	1
Total Number of Cores:	2
Graphics: Intel UHD Graphics 617
Memory:	8 GB
Memory: Size: 4 GB, Type:LPDDR3, Speed: 2133 MHz
Memory: Size: 4 GB, Type:LPDDR3, Speed: 2133 MHz

Kieran
